employees/serializers.py

```diff
@@ -48,8 +50,29 @@ class Meta:
         model = Report
         fields = ("url", "date", "project", "author", "task_activities", "description", "work_hours")
 
-    def to_representation(self, instance: Report) -> Report:
+    def to_representation(self, instance: Report) -> OrderedDict:
         data = super().to_representation(instance)
         if isinstance(self.instance, Report):
             data["work_hours"] = self.instance.work_hours_str
         return data
+
+    def validate(self, data: OrderedDict) -> OrderedDict:
+        data = super().validate(data)
+        if "date" in data:
```

Comment:
Ok, since it seems that the serializer is never created with `partial` flag (https://www.django-rest-framework.org/api-guide/serializers/#partial-updates), it's safe to assume, that `"date"` is always present in `data`. So, please remove the `if`.

Reply:
Based on that, I additionally changed `data.get("work_hours", 0)` to just `data["work_hours"]`